client/state/data-layer/wpcom-http/utils.js

```diff
@@ -73,17 +100,21 @@ const defaultOptions = {
  *   onSuccess  :: ReduxStore -> Action -> Dispatcher -> ResponseData
  *   onError    :: ReduxStore -> Action -> Dispatcher -> ErrorData
  *   onProgress :: ReduxStore -> Action -> Dispatcher -> ProgressData
+ *   fromApi    :: ResponseData -> [ Boolean, Data ]
  *
  * @param {Function} initiator called if action lacks response meta; should create HTTP request
  * @param {Function} onSuccess called if the action meta includes response data
  * @param {Function} onError called if the action meta includes error data
  * @param {Object} options configures additional dispatching behaviors
- + @param {Function} [options.middleware] runs before the dispatch itself
+ + @param {Function} [options.fromApi] maps between API data and Calypso data
```

Comment:
It would be nice to explicitly state the return of `fromApi` is something like `[ isValidBySchemaAndTransformedWithoutErrors, maybeTransformedData ]`.

Reply:
do you know a good way apart from creating a new `@typedef` or something?

Reply:
moot point after the changes @jsnajdr introduced by `throw`ing instead of returning the pair